utils/staking-miner/src/rpc.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! RPC related types and helpers.
+
+use super::*;
+use jsonrpsee::{
+	core::{Error as RpcError, RpcResult},
+	proc_macros::rpc,
+};
+use pallet_transaction_payment::RuntimeDispatchInfo;
+use sc_transaction_pool_api::TransactionStatus;
+use sp_core::{storage::StorageKey, Bytes};
+use sp_version::RuntimeVersion;
+use std::{future::Future, time::Duration};
+
+#[derive(frame_support::DebugNoBound, thiserror::Error)]
+pub(crate) enum RpcHelperError {
+	JsonRpsee(#[from] jsonrpsee::core::Error),
+	Codec(#[from] codec::Error),
+}
+
+impl std::fmt::Display for RpcHelperError {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		<RpcHelperError as std::fmt::Debug>::fmt(self, f)
+	}
+}
+
+#[rpc(client)]
+pub trait RpcApi {
+	/// Fetch system name
+	#[method(name = "system_chain")]
+	async fn system_chain(&self) -> RpcResult<String>;
+
+	/// Fetch a storage key
+	#[method(name = "state_getStorage")]
+	async fn storage(&self, key: &StorageKey, hash: Option<&Hash>) -> RpcResult<Option<Bytes>>;
+
+	/// Fetch the runtime version
```

Comment:
```suggestion
	/// Fetch the runtime version.
```

nit: punctuation in docs are not consistent.